pandas/core/dtypes/dtypes.py

```diff
@@ -2,12 +2,70 @@
 
 import re
 import numpy as np
+from collections import OrderedDict
 from pandas import compat
 from pandas.core.dtypes.generic import ABCIndexClass, ABCCategoricalIndex
 
 from .base import ExtensionDtype, _DtypeOpsMixin
 
 
+class Registry(object):
+    """ Registry for dtype inference
+
+    We can directly construct dtypes in pandas_dtypes if they are
+    a type; the registry allows us to register an extension dtype
+    to try inference from a string or a dtype class
+
+    These are tried in order for inference.
+    """
+    dtypes = OrderedDict()
+
+    @classmethod
+    def register(self, dtype, constructor=None):
+        """
+        Parameters
+        ----------
+        dtype : PandasExtension Dtype
+        """
+        if not issubclass(dtype, (PandasExtensionDtype, ExtensionDtype)):
```

Comment:
Shouldn't ExtensionDtype be enough? (our internals one subclass both I think?)

Reply:
not yet, are extension dtypes are not all ExtensiDtypes yet (this is why we have the whole PandasExtensionDtype bizness)